source/watchdog.py

```diff
@@ -214,7 +213,7 @@ def _crashHandler(exceptionInfo):
 	log.info(f"Listing stacks for Python threads:\n{stacks}")
 
 	log.info("Restarting due to crash")
-	core.restart()
+	core.restartUnsafely()  # if NVDA has crashed we cannot rely on it to exit safely

```

Comment:
It might be more accurate to say:
"we can't rely on the queue handler to do the call to restart NVDA"

Reply:
```suggestion
	core.restartUnsafely()  # if NVDA has crashed we cannot rely on the queue handler to start the new NVDA instance
```

Reply:
If we are confident that this is the only problem we should make the change by performing a direct call to _doShutdown instead of queueing it.